### DISCUSSION TITLE
`EffectPublisher.operation` is private

### DISCUSSION THREAD

**Discussion by ivanmoskalev** [pointfreeco/swift-composable-architecture#1915] [Q&A]
### Description

Good afternoon! Not exactly a bug.

I'm trying to implement a reducer that would work akin to the `_SignpostReducer` — but logging the trace to something like Sentry or Firebase Performance. 

For this I need access to `EffectPublisher.operation`, which is declared as `internal`. 

It would be possible to implement the tracing reducer either:
1. By opening up access to the variable.
2. By generalizing the `_SignpostReducer` so that it can be constructed with closures that would get called [before and/or] after the effects get executed. 

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

**Comment by mbrandonw** [comment]
Hi @ivanmoskalev, while you can't provide the exact implementation that the `effectSignpost` method does, you can still accomplish what you want since `Effect` still does conform to the `Publisher` protocol. You can use the `handleEvents` method from Combine to instrument all of the events of the effect, like [done here](https://github.com/pointfreeco/swift-composable-architecture/blob/1b998c43b3b3505876db6f66e03243dd2453e93a/Sources/ComposableArchitecture/Reducer/AnyReducer/AnyReducerSignpost.swift#L103-L123).

I'm going to convert this to a discussion since it isn't really an issue with the library, but feel free to ask any follow up questions.
